src/Implement/MainManager.cpp

```diff
@@ -0,0 +1,86 @@
+#include "MainManager.h"
+
+MainManager *MainManager::only_mainManager = nullptr;
+
+namespace {
+void SigHandle(int sig, siginfo_t *info, void *ctx) {
+  if (sig == SIGUSR1) {
+    mainManager.RemoveByPid(info->si_pid);
+  }
+}
+}  // namespace
+
+void MainManager::SetRequestHandler(const RequestHandler &request_handler) {
+  this->request_handler = request_handler;
+}
+
+void MainManager::InitInstance(const uint16_t &port_num) {
+  this->port_num = port_num;
+  // when child process ends, it'll exit itself and need to notify parent
+  // process to remove it from socket_map child process will end with calling
+  // EndRequest()
+  struct sigaction act;
+  act.sa_sigaction = SigHandle;
+  sigemptyset(&act.sa_mask);
+  act.sa_flags = SA_SIGINFO;
+  sigaction(SIGUSR1, &act, NULL);
```

Comment:
Where did you add the comment on SIGUSR1?

Reply:
Done